nsone/rest/transport/basic.py

```diff
@@ -62,11 +62,11 @@ def handleProblem(code, resp, msg):
             resp = opener.open(request, timeout=self._timeout)
         except HTTPError as e:
             resp = e
-            body = resp.read()
         except Exception as e:
             body = '"Service Unavailable"'
             resp = HTTPError(url, 503, body, headers, None)
         finally:
+            body = resp.read()
```

Comment:
Thank you for the pull request! Unfortunately, `body = resp.read()` will raise an `AttributeError` if an exception is raised that is not an instance of `HTTPError`. I think it would be better to add `body = resp.read()` after `opener.open`:

```
        try:
            resp = opener.open(request, timeout=self._timeout)
            body = resp.read()
        except HTTPError as e:
            resp = e
            body = resp.read()
        except Exception as e:
            resp = HTTPError(url, 503, body, headers, None)
            body = '"Service Unavailable"'
        finally:
            if resp.code != 200:
                handleProblem(resp.code, resp, body)
```